pkg/kubelet/dockertools/manager.go

```diff
@@ -1013,13 +1013,19 @@ func (dm *DockerManager) getSecurityOpt(pod *api.Pod, ctrName string) ([]string,
 		return nil, nil
 	}
 
-	if !strings.HasPrefix(profile, "localhost") {
+	if !strings.HasPrefix(profile, "localhost/") {
 		return nil, fmt.Errorf("unknown seccomp profile option: %s", profile)
 	}
 
-	file, err := ioutil.ReadFile(filepath.Join(dm.seccompProfileRoot, strings.TrimPrefix(profile, "localhost/")))
+	name := strings.TrimPrefix(profile, "localhost/")
+	cleanName := strings.TrimPrefix(path.Clean("/"+name), "/")
+	if name != cleanName {
+		return nil, fmt.Errorf("invalid seccomp profile name: %s", name)
```

Comment:
I talked with @pmorie on irc an we think ideally we'd have a helper function to do this, as it's needed in multiple places. There's already private validation code at https://github.com/kubernetes/kubernetes/blob/5288a255f486e1c2c4fce109d369c52d6ed456e5/pkg/api/validation/validation.go#L742 to check for attempts to escape a base directory.

Given that we probably shouldn't change the validation code during the 1.3 code freeze, it's probably best to copy the logic from validateSubPath into here and put a TODO to create a helper and unify this code and the other places we need this logic after 1.3 is out.